src/script/descriptor.cpp

```diff
@@ -345,10 +345,11 @@ class BIP32PubkeyProvider final : public PubkeyProvider
         key_out = final_extkey.pubkey;
 
         if (write_cache) {
-            write_cache->CacheDerivedExtPubKey(m_expr_index, pos, final_extkey);
             // Only cache parent if there is any unhardened derivation
             if (m_derive != DeriveType::HARDENED) {
                 write_cache->CacheParentExtPubKey(m_expr_index, parent_extkey);
+            } else if (child_info_out.path.size() > 0) {
```

Comment:
e4262c45655b4f79ddfd9905db66dc44b8f5148c doesn't compile (`child_info_out`  is  renamed in In d7b241109f74b3c1c407929930e7d035cdcf60c9 `to `child_info_out`)

Reply:
Fixed